tests/optim/test_oss.py

```diff
@@ -501,7 +501,7 @@ def run_test_reproducibility(rank, world_size, tempfile_name):
     loss_fn = torch.nn.L1Loss()
     loss_fn.to(device)
 
-    optimizer = optim.OSS(model.parameters(), optim=torch.optim.RMSprop, lr=0.1)
+    optimizer = optim.OSS(model.parameters(), optim=torch.optim.RMSprop, lr=0.1, broadcast_fp16=broadcast_fp16)
```

Comment:
Can we do a weight check as well? I think you have a test for that as well if i remember correctly.

Reply:
the bucket test is doing that, here I think that the comparison would be not completely trivial, because some accuracy is lost when doing fp32->fp16->fp32, so the result will not be strictly the same as not doing it. The bucket unit test does check that the results are the same as doing this conversion chain outside of the bucket

Reply:
discussed offline: Since the weights may not be exactly equal we want to know what the tolerance diff value could be.